app/src/main/java/org/pesmypetcare/mypetcare/services/VetVisitsManagerService.java

```diff
@@ -0,0 +1,57 @@
+package org.pesmypetcare.mypetcare.services;
+
+import org.pesmypetcare.mypetcare.features.pets.Pet;
+import org.pesmypetcare.mypetcare.features.pets.VetVisit;
+import org.pesmypetcare.mypetcare.features.pets.VetVisitAlreadyExistingException;
+import org.pesmypetcare.mypetcare.features.users.User;
+import org.pesmypetcare.usermanagerlib.datacontainers.DateTime;
+
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * @author Xavier Campos
+ */
+public interface VetVisitsManagerService {
+
+    /**
+     * Obtains all the vet visits for the given pet.
+     * @param user The owner of the pet
+     * @param pet The pet from which we want to obtain all the vet visits
+     * @return A list containing all the vet visits of the pet
+     */
+    List<VetVisit> findVetVisitsByPet(User user, Pet pet) throws ExecutionException, InterruptedException;
+
+    /**
+     * Creates and adds a new vet visit to the given pet.
+     * @param user The owner of the pet
+     * @param pet The pet where the vet visits has to be added
+     * @param vetVisit The vet visit that has to be added to the pet
+     */
+    void createVetVisit(User user, Pet pet, VetVisit vetVisit) throws VetVisitAlreadyExistingException, ExecutionException, InterruptedException;
+
+    /**
+     * Deletes the vet visit from the given pet.
+     * @param user The owner of the pet
+     * @param pet The pet from where the vet visit has to be removed
+     * @param vetVisit The vet visit that has to be removed from the pet
+     */
+    void deleteVetVisit(User user, Pet pet, VetVisit vetVisit) throws ExecutionException, InterruptedException;
+
+    /**
+     * Updates the key of the visit date for the given visit date.
+     * @param user The owner of the pet
+     * @param pet The pet from which we want to update the visit
+     * @param newDate The new date to set to the visit
+     * @param visitDate The current date of the visit
+     */
+    void updateVetVisitKey(User user, Pet pet, String newDate, DateTime visitDate) throws ExecutionException, InterruptedException;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line is longer than 120 characters (found 131).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5493224)